### PR TITLE
RDK-38411: Unit test for RDKShell

### PR DESCRIPTION
**Pull request by zhilarasi**

### CODE REVIEW

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -207,14 +217,16 @@ jobs:
...
           -include ${{github.workspace}}/rdkservices/Tests/mocks/devicesettings.h
           -include ${{github.workspace}}/rdkservices/Tests/mocks/FrameBuffer.h
           -include ${{github.workspace}}/rdkservices/Tests/mocks/Iarm.h
           -include ${{github.workspace}}/rdkservices/Tests/mocks/Rfc.h
           -include ${{github.workspace}}/rdkservices/Tests/mocks/RBus.h
           -include ${{github.workspace}}/rdkservices/Tests/mocks/Telemetry.h
           -include ${{github.workspace}}/rdkservices/Tests/mocks/Udev.h
-          --coverage -Wall -Werror -Wno-unused-parameter -Wno-unused-result -Wl,-wrap,system -Wl,-wrap,popen -Wl,-wrap,syslog
+          -include ${{github.workspace}}/rdkservices/Tests/mocks/rdkshell.h
+          --coverage -Wall -Werror -Wno-unused-parameter -Wno-unused-result -Wno-unused-variable -Wl,-wrap,system -Wl,-wrap,popen -Wl,-wrap,syslog
```
Comment (npoltorapavlo) [inline .github/workflows/unit-tests.yml:229]: please revert `-Wno-unused-variable` and fix the code instead

**Review by npoltorapavlo — CHANGES_REQUESTED** [review]

File: Tests/tests/test_RDKShell.cpp [hunk 2]
```diff
@@ -0,0 +1,649 @@
...
+		      scaleY = 0.5;
+                      return true;
+                }));
+
+  EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("getScale"), _T("{\"client\": \"org.rdk.Netflix\","
+                                                                                             "\"callsign\": \"org.rdk.Netflix\"}"), response));
+
+
+}
+
```
Comment (npoltorapavlo) [inline Tests/tests/test_RDKShell.cpp:649]: shouldn't you check JSON-RPC response in JSON-RPC tests?

**Review by npoltorapavlo — CHANGES_REQUESTED** [review]

File: Tests/tests/test_RDKShell.cpp (hunk 2 above)
Comment (zhilarasi) [reply Tests/tests/test_RDKShell.cpp:649]: i will add that

File: Tests/mocks/rdkshell.h [hunk 3]
```diff
@@ -0,0 +1,584 @@
...
+            {
+		    return getInstance().impl->getVirtualResolution(client, virtualWidth, virtualHeight);
+            }
+            static bool setVirtualResolution(const std::string& client, const uint32_t virtualWidth, const uint32_t virtualHeight)
+            {
+		    return getInstance().impl->setVirtualResolution(client, virtualWidth, virtualHeight);
+            }
+            static bool enableVirtualDisplay(const std::string& client, const bool enable){return true;}
+            static bool getVirtualDisplayEnabled(const std::string& client, bool &enabled){return true;}
+            static bool getLastKeyPress(uint32_t &keyCode, uint32_t &modifiers, uint64_t &timestampInSeconds){return true;}
```
Comment (npoltorapavlo) [inline Tests/mocks/rdkshell.h:533]: please move implementations to the mock

File: Tests/mocks/rdkshell.h [hunk 4]
```diff
@@ -0,0 +1,584 @@
...
+    #define RDKSHELL_APPLICATION_MIME_TYPE_HTML "application/html"
+    #define RDKSHELL_APPLICATION_MIME_TYPE_LIGHTNING "application/lightning"
+
+    enum class ApplicationState
+    {
+        Unknown,
+        Running,
+        Suspended,
+        Stopped
+    };
```
Comment (npoltorapavlo) [inline Tests/mocks/rdkshell.h:302]: can this be removed?

File: Tests/mocks/rdkshell.h [hunk 5]
```diff
@@ -0,0 +1,584 @@
...
+      static void resolveWaitingEasterEggs(){return;}
+      static void addEasterEgg(std::vector<RdkShellEasterEggKeyDetails>& details, std::string name, uint32_t timeout, std::string actionJson){return;}
+      static void removeEasterEgg(std::string name){return;}
+      static void getEasterEggs(std::vector<RdkShellEasterEggDetails>& easterEggs){return;}
+      static void initialize(){return;}
+      static void run(){return;}
+      static void update(){return;}
+      static void draw(){return;}
+      static void deinitialize(){return;}
+      static double seconds(){return 0.0;}
```
Comment (npoltorapavlo) [inline Tests/mocks/rdkshell.h:268]: please move implementations to the mock

File: Tests/mocks/rdkshell.h [hunk 6]
```diff
@@ -0,0 +1,584 @@
...
+
+    class Logger
+    {
+        public:
+            static void log(LogLevel level, const char* format, ...){return;}
+            static void logLevel(std::string& level)
+           {
+                   std::cout << "getcall" << std::endl;
+                   level = sLogLevel;
+           }
```
Comment (npoltorapavlo) [inline Tests/mocks/rdkshell.h:214]: can this be removed?

File: Tests/mocks/rdkshell.h [hunk 7]
```diff
@@ -0,0 +1,584 @@
...
+            static void enableFlushing(bool enable){return;}
+            static bool isFlushingEnabled(){return true;}
+       private:
+            static LogLevel sLogLevel;
+            static bool sFlushingEnabled;
```
Comment (npoltorapavlo) [inline Tests/mocks/rdkshell.h:219]: can this be removed?

File: Tests/mocks/rdkshell.h [hunk 8]
```diff
@@ -0,0 +1,584 @@
...
+    static const char* logLevelStrings[] =
+    {
+      "DEBUG",
+      "INFO",
+      "WARN",
+      "ERROR",
+      "FATAL"
+    };
+
+    static const int numLogLevels = sizeof(logLevelStrings)/sizeof(logLevelStrings[0]);
```
Comment (npoltorapavlo) [inline Tests/mocks/rdkshell.h:231]: can this be removed?

File: Tests/mocks/rdkshell.h [hunk 9]
```diff
@@ -0,0 +1,584 @@
...
+    const std::string RDKSHELL_EVENT_POWER_KEY = "onPowerKey";
+    const std::string RDKSHELL_EVENT_KEY = "onKeyEvent";
+    const std::string RDKSHELL_EVENT_SIZE_CHANGE_COMPLETE = "onSizeChangeComplete";
+   enum LogLevel {
+        Debug,
+        Information,
+        Warn,
+        Error,
+        Fatal
+    };
```
Comment (npoltorapavlo) [inline Tests/mocks/rdkshell.h:204]: can this be removed?

File: Tests/mocks/rdkshell.h [hunk 10]
```diff
@@ -0,0 +1,584 @@
...
+    const std::string RDKSHELL_EVENT_USER_INACTIVE = "onUserInactive";
+    const std::string RDKSHELL_EVENT_DEVICE_LOW_RAM_WARNING = "onDeviceLowRamWarning";
+    const std::string RDKSHELL_EVENT_DEVICE_CRITICALLY_LOW_RAM_WARNING = "onDeviceCriticallyLowRamWarning";
+    const std::string RDKSHELL_EVENT_DEVICE_LOW_RAM_WARNING_CLEARED = "onDeviceLowRamWarningCleared";
+    const std::string RDKSHELL_EVENT_DEVICE_CRITICALLY_LOW_RAM_WARNING_CLEARED = "onDeviceCriticallyLowRamWarningCleared";
+    const std::string RDKSHELL_EVENT_ANIMATION = "onAnimation";
+    const std::string RDKSHELL_EVENT_EASTER_EGG = "onEasterEgg";
+    const std::string RDKSHELL_EVENT_POWER_KEY = "onPowerKey";
+    const std::string RDKSHELL_EVENT_KEY = "onKeyEvent";
+    const std::string RDKSHELL_EVENT_SIZE_CHANGE_COMPLETE = "onSizeChangeComplete";
```
Comment (npoltorapavlo) [inline Tests/mocks/rdkshell.h:197]: can this be removed?

File: Tests/mocks/rdkshell.h [hunk 11]
```diff
@@ -0,0 +1,584 @@
...
+          virtual void onApplicationLaunched(const std::string& client) {}
+          virtual void onApplicationConnected(const std::string& client) {}
+          virtual void onApplicationDisconnected(const std::string& client)  {}
+          virtual void onApplicationTerminated(const std::string& client) {}
+          virtual void onApplicationFirstFrame(const std::string& client) {}
+          virtual void onApplicationSuspended(const std::string& client) {}
+          virtual void onApplicationResumed(const std::string& client) {}
+          virtual void onApplicationActivated(const std::string& client) {}
+          virtual void onUserInactive(const double minutes) {}
+          virtual void onDeviceLowRamWarning(const int32_t freeKb, const int32_t availableKb, const int32_t usedSwapKb) {}
```
Comment (npoltorapavlo) [inline Tests/mocks/rdkshell.h:169]: please move implementations to the mock

File: Tests/mocks/rdkshell.h [hunk 12]
```diff
@@ -0,0 +1,584 @@
...
+            int64_t toInteger64() const{return 0;}
+            uint8_t toUnsignedInteger8() const{return 0;}
+            uint32_t toUnsignedInteger32() const{return 0;}
+            uint64_t toUnsignedInteger64() const{return 0;}
+            float toFloat() const{return 0.0;}
+            double toDouble() const{return 0.0;}
+            std::string toString() const{return "a";}
+            void* toVoidPointer() const{return NULL;}
+
+            RdkShellData& operator=(bool value){return *this;}
```
Comment (npoltorapavlo) [inline Tests/mocks/rdkshell.h:135]: please move implementations to the mock

File: Tests/mocks/rdkshell.h [hunk 13]
```diff
@@ -0,0 +1,584 @@
...
+        uint32_t    unsignedInteger32Data;
+        uint64_t    unsignedInteger64Data;
+        float       floatData;
+        double      doubleData;
+        std::string* stringData;
+        void*       pointerData;
+        RdkShellDataInfo() {}
+        ~RdkShellDataInfo() {}
+    };
+
```
Comment (npoltorapavlo) [inline Tests/mocks/rdkshell.h:91]: can this be removed?

File: Tests/mocks/rdkshell.h [hunk 14]
```diff
@@ -0,0 +1,584 @@
...
+#include <unistd.h>
+#include <string.h>
+#include <stdio.h>
+#include <iostream>
+#include <ctime>
+#include <sys/types.h>
+#include <thread>
+#include <mutex>
+#include <memory>
+#define RDKSHELL_SPLASH_SCREEN_FILE_CHECK "/tmp/.rdkshellsplash"
```
Comment (npoltorapavlo) [inline Tests/mocks/rdkshell.h:16]: can this be removed?

File: Tests/mocks/rdkshell.h [hunk 15]
```diff
@@ -0,0 +1,584 @@
...
+#include <string>
+#include <map>
+#include <vector>
+#include <unistd.h>
+#include <string.h>
+#include <stdio.h>
+#include <iostream>
+#include <ctime>
+#include <sys/types.h>
+#include <thread>
```
Comment (npoltorapavlo) [inline Tests/mocks/rdkshell.h:13]: can this be removed?

**Review by npoltorapavlo — CHANGES_REQUESTED** [review]

**Pull request closed without merging**